src/Sql/Sql/ManagedDatabase Backup/Cmdlet/AzureSqlManagedDatabaseBackupCmdletBase.cs

```diff
@@ -0,0 +1,203 @@
+        /// </summary>
+        [Parameter(ParameterSetName = PolicyByResourceServerDatabaseSet,
+            Mandatory = true,
+            Position = 2,
+            HelpMessage = "The name of the Azure SQL Instance Database to retrieve backups for.")]
+        [ResourceNameCompleter("Microsoft.Sql/managedInstances/databases", "ResourceGroupName", "InstanceName")]
+        [ValidateNotNullOrEmpty]
+        public virtual string DatabaseName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the deletion date of the database to use.
+        /// </summary>
+        [Parameter(ParameterSetName = PolicyByResourceServerDatabaseSet,
+            Mandatory = false,
+            Position = 3,
+            HelpMessage = "The deletion date of the Azure SQL Instance Database to retrieve backups for, with millisecond precision (e.g. 2016-02-23T00:21:22.847Z)")]
+        [ValidateNotNullOrEmpty]
+        public virtual DateTime? DeletionDate { get; set; }
+        /// <summary>
+        /// Initializes the adapter
+        /// </summary>
+        /// <returns></returns>
+        protected override AzureSqlManagedDatabaseBackupAdapter InitModelAdapter()
+        {
+            return new AzureSqlManagedDatabaseBackupAdapter(DefaultProfile.DefaultContext);
+        }
+
+        public override void ExecuteCmdlet()
+        {
+            if (AzureInstanceDatabaseObject != null)
+            {
+                this.ResourceGroupName = AzureInstanceDatabaseObject.ResourceGroupName;
+                this.InstanceName = AzureInstanceDatabaseObject.ManagedInstanceName;
+                this.DatabaseName = AzureInstanceDatabaseObject.Name;
+                if (AzureInstanceDatabaseObject is AzureSqlDeletedManagedDatabaseBackupModel)
+                {
+                    this.DeletionDate = ((AzureSqlDeletedManagedDatabaseBackupModel)AzureInstanceDatabaseObject).DeletionDate;
+                }
+            }
+            else if (!string.IsNullOrEmpty(ResourceId))
+            {
+                ParseResourceId();
+            }
+
+            base.ExecuteCmdlet();
+        }
+
+        /// <summary>
+        /// Helper method to parse resourceGroupName, serverName, databaseName from resourceId
+        /// Ideally this would utilize Microsoft.Azure.Management.Internal.Resources.Utilities.Models.ResourceIdentifier
+        /// However, that class is not setup to handle resources that have an ancestry higher than just parent level.
+        /// That class could recursively create a parent ResourceIdentifier on a __get__ ParentResource,
+        /// rather than just a string, so a consumer could work all the way to the root resource easily.
+        /// Leave as a TODO for now, as many other cmdlets consume that class, and I will work on it in separate change.
+        /// </summary>
+        internal void ParseResourceId()
+        {
+            string[] tokens = ResourceId.Split(new char[] { '/' }, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
That is not really what I need. Do you want me to add my logic to the ResourceIdentifier?

Reply:
Hmm okay, I see where this does get more complex. Since this is a rather special case, let's leave this rather than adding specific case logic to ResourceIdentifier.